### ISSUE TITLE
MissingPrimaryColumnError in non-existing entity, and marked with @PrimaryColumn

### ISSUE THREAD

**Issue by Stdev17** [typeorm/typeorm#5644]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[x] `0.2.22` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

=======

Since an entity in TypeORM must have a primary column, I've got the error at first time. So I quickly fix the entity with \@PrimaryColumn. However, same error occured regardless of a primary column does exist or not. Finally I threw away that file and ran a test, but the error said THE entity I had deleted does not have a primary column. I dropped the database and created it again. same result.

I believe that one of the concept of ORM is automatically reform the schema without having to deal with those DDL statements. In this case, TypeORM doesn't seem to changed any schema of the written models and just got stuck. show tables; shows me Empty set (0.00 sec). I wonder how can I make TypeORM to do their job.

The model with problem is...
```
import { Column, Entity, ManyToOne, PrimaryColumn } from "typeorm";
import { Profile } from "./Profile";

@Entity()
export class ProfileConfig {
  @PrimaryColumn()
  configId!: string;
  @ManyToOne(
    () => Profile,
    charProfileItem => charProfileItem.charProfileItem
  )
  charProfileItem?: Profile;
  @Column({ default: null, nullable: true, type: String })
  charProfileWhitelist?: string | null;
}
```

I ran my test with simple code under this...
```
import { Connection, createConnection } from "typeorm";

describe("Sample Test", () => {
  let typeormConnection: Connection;

  beforeAll(async () => {
    typeormConnection = await createConnection();
    await typeormConnection.synchronize();
  });

  it("should be run with no error", () => {
    expect({}).toBeInstanceOf(Object);
  });

  it("should connect to DBMS", () => {
    expect(typeormConnection.isConnected).toBeTruthy();
  });

  afterAll(async () => {
    await typeormConnection.close();
  });
});
```

**Comment by Stdev17** [comment]
This is the call stack. I investigated EntityMetaValidator for where the bug kicks in.

```
MissingPrimaryColumnError: Entity "ProfileConfig" does not have a primary column. Primary column is required to have in all your entities. Use @PrimaryColumn decorator to add a primary column to your entity.

      at new MissingPrimaryColumnError (src/error/MissingPrimaryColumnError.ts:9:9)
      at EntityMetadataValidator.Object.<anonymous>.EntityMetadataValidator.validate (src/metadata-builder/EntityMetadataValidator.ts:57:19)
      at src/metadata-builder/EntityMetadataValidator.ts:45:56
          at Array.forEach (<anonymous>)
      at EntityMetadataValidator.Object.<anonymous>.EntityMetadataValidator.validateMany (src/metadata-builder/EntityMetadataValidator.ts:45:25)
      at Connection.Object.<anonymous>.Connection.buildMetadatas (src/connection/Connection.ts:526:33)
      at Connection.<anonymous> (src/connection/Connection.ts:191:18)
      at step (node_modules/tslib/tslib.js:136:27)
      at Object.next (node_modules/tslib/tslib.js:117:57)
      at fulfilled (node_modules/tslib/tslib.js:107:62)
```

Simply digging in entityMetaData, where the entity metadata of the written models is recorded, primaryColumns of an entity are represented as [ [ColumnMetadata] ]. Those EntityMetaDatas are fixed already, and I can't change it by editing the code. In this case, primaryColumns of ProfileConfig is [].

The metadatas are built in this lines of code. This seems to be not hand-written code, but transpiled by tsc.
```
    /**
     * Builds entity metadatas for the given classes or directories.
     */
    ConnectionMetadataBuilder.prototype.buildEntityMetadatas = function (entities) {
        // todo: instead we need to merge multiple metadata args storages
        var _a = tslib_1.__read(OrmUtils_1.OrmUtils.splitClassesAndStrings(entities || []), 2), entityClassesOrSchemas = _a[0], entityDirectories = _a[1];
        var entityClasses = entityClassesOrSchemas.filter(function (entityClass) { return (entityClass instanceof EntitySchema_1.EntitySchema) === false; });
        var entitySchemas = entityClassesOrSchemas.filter(function (entityClass) { return entityClass instanceof EntitySchema_1.EntitySchema; });
        var allEntityClasses = tslib_1.__spread(entityClasses, DirectoryExportedClassesLoader_1.importClassesFromDirectories(this.connection.logger, entityDirectories));
        allEntityClasses.forEach(function (entityClass) {
            if (entityClass instanceof EntitySchema_1.EntitySchema) {
                entitySchemas.push(entityClass);
                allEntityClasses.slice(allEntityClasses.indexOf(entityClass), 1);
            }
        });
        var decoratorEntityMetadatas = new EntityMetadataBuilder_1.EntityMetadataBuilder(this.connection, index_1.getMetadataArgsStorage()).build(allEntityClasses);
        var metadataArgsStorageFromSchema = new EntitySchemaTransformer_1.EntitySchemaTransformer().transform(entitySchemas);
        var schemaEntityMetadatas = new EntityMetadataBuilder_1.EntityMetadataBuilder(this.connection, metadataArgsStorageFromSchema).build();
        return tslib_1.__spread(decoratorEntityMetadatas, schemaEntityMetadatas);
    };
```

I checked out entityDirectories and... bingo. those were [ 'dist/entity/*.ts', 'dist/entity/*.js' ]. This is actually the same with a key, entities in ormconfig.js. If original typescript models were not built by tsc, It doesn't work.

TLDR; this was not actually a bug. however, now I've got a new error!
